### DISCUSSION TITLE
pymupdf4llm and page 0 counting upwards

### DISCUSSION THREAD

**Discussion by easyname889** [pymupdf/PyMuPDF#3730] [Looking for help]
Hi and thanks for a great library!

Im running 
```
import pymupdf4llm

md_text = pymupdf4llm.to_markdown("input.pdf")

# now work with the markdown text, e.g. store as a UTF8-encoded file
import pathlib
pathlib.Path("output.md").write_bytes(md_text.encode())
```
But the pages starts at 0 which makes the reference system i built furter down my data pipe a mess.

I can add a loop over the files in the folder and do a +1 on every page number but then i need to do the same in the extracted .md files as well for the images. Is there a more gracious way to tell pymupdf4llm to start counting page 1 as one, instead of 0? That would save me a lot of post processing.

Tanks

**Comment by JorjMcKie** [comment]
pymupdf4llm here follows the same convention as pymupdf and literally **_all_** other libraries dealing with document management.
Only page numbers starting at 0 have a chance to treat them as a sequence in Python with all the advantages associated with slicing, range support and more. E.g. only this convention makes clear what `doc[-1]` means: the last page.
So there literally is no way to change that.
However: if you request `page_chunks=True`, every page's text and other information is represented as a dictionary inside a list of the other page dictionaries.
A page "chunk" dictionary has the key `"page"` which is the 1-based page number. IAW page 0 has `"page": 1` in that dictionary.
